hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -144,6 +186,15 @@ class MergeOnReadSnapshotRelation(sqlContext: SQLContext,
       fsView.getLatestMergedFileSlicesBeforeOrOn(relativePath, queryTimestamp).iterator().asScala.toSeq
     }
   }
+
+  protected def pruneSchemaForMergeSkipping(requiredSchema: HoodieTableSchema): HoodieTableSchema = {
+    val mandatoryFieldNames = mandatoryFields.map(fieldName => HoodieAvroUtils.getRootLevelFieldName(fieldName))
+    val prunedStructSchema = StructType(
+      requiredSchema.structTypeSchema.fields.filterNot(f => mandatoryFieldNames.contains(f.name))
+    )
```

Comment:
We should not exclude the root fields from `mandatoryFieldNames` since the query columns may also contain the same root field.

Reply:
Yeah, realized that as well. This has already been addressed.